integration/flink/shared/src/main/java/io/openlineage/flink/utils/CassandraUtils.java

```diff
@@ -37,4 +42,45 @@ public static Optional<Table> extractTableAnnotation(Class pojo) {
 
     return Optional.empty();
   }
+
+  public static Optional<String> findNamespaceFromBuilder(
+      Optional<ClusterBuilder> clusterBuilderOpt) {
+    try {
+      if (clusterBuilderOpt.isPresent()) {
+        Cluster cluster = clusterBuilderOpt.get().getCluster();
+        if (cluster != null) {
+          Optional<Object> managerOpt =
+              WrapperUtils.<Object>getFieldValue(Cluster.class, cluster, "manager");
+          if (managerOpt.isPresent()) {
+            Class managerClass = Class.forName(CASSANDRA_MANAGER_CLASS);
+            Optional<List<Object>> endpointsOpt =
+                WrapperUtils.<List<Object>>getFieldValue(
+                    managerClass, managerOpt.get(), "contactPoints");
+            return CassandraUtils.convertToNamespace(endpointsOpt);
+          }
+        }
+      }
+    } catch (ClassNotFoundException e) {
+      log.error("Failed load class required to infer the Cassandra namespace name", e);
+    }
+
+    return Optional.of("");
+  }
+
+  public static ClusterBuilder createClusterBuilder(String contactPoints) {
+    return new ClusterBuilder() {
+      @Override
+      protected Cluster buildCluster(Cluster.Builder builder) {
+        return builder.addContactPoint(contactPoints).build();
+      }
+    };
+  }
+
+  private static Optional<String> convertToNamespace(Optional<List<Object>> endpointsOpt) {
+    if (endpointsOpt.isPresent() && !endpointsOpt.isEmpty()) {
+      return Optional.of("cassandra://" + endpointsOpt.get().get(0).toString().split("/")[1]);
```

Comment:
this can still fail on `endpointsOpt.get().get(0)` being null or `endpointsOpt.get().get(0).toString().split("/")` having single element.